### PR TITLE
Admin dashboard

### PR DESCRIPTION
**Pull request by kristi-h**
Added a search bar with customized filtering for all columns.

I tried to use the built in FilterModel from mui data x grid, but the free community model only allowed for one filter, so scrapped that.

### CODE REVIEW

**Comment by DalySoftware** [comment]
Doesn't this reimplement the built in filter?
https://mui.com/x/react-data-grid/filtering/quick-filter/

If you wanted to try rolling your own fair enough

File: server/src/controllers/appointments.controller.js
Comment (DalySoftware) [inline server/src/controllers/appointments.controller.js:1]: Looks like a lots of bad merges in this file. I assume you have these in hand

File: client/components/admin_dashboard/Grid.jsx [hunk 1]
```diff
@@ -88,27 +90,59 @@ const Toolbar = () => (
...
+      columns.some((col) => {
+        const value = row[col.field];
+
+        if (col.field === "dateCreated") {
+          const formattedDate = formatDateCreated(new Date(value));
+          return formattedDate.toLowerCase().includes(searchText);
+        }
+        if (col.field === "timeRange") {
+          const formattedTimeRange = formatTimeRange(value);
+          return formattedTimeRange.toLowerCase().includes(searchText);
```
[the remark this answers is not in the document]
Comment (kristi-h) [reply client/components/admin_dashboard/Grid.jsx:107]: right!

File: client/components/admin_dashboard/SearchBar.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <Box sx={{ display: "flex", justifyContent: "center", width: "100%" }}>
+      <TextField
+        label="Search"
+        variant="outlined"
+        margin="normal"
+        value={searchText}
+        onChange={handleSearch}
+        sx={{
+          width: "55%",
```
[the remark this answers is not in the document]
Comment (kristi-h) [reply client/components/admin_dashboard/SearchBar.jsx:19]: ah ok, I'll fix this

File: server/src/utils/emailHelper.js [hunk 3]
```diff
@@ -99,8 +99,9 @@ const sendEmail = async ({ toAddress, subject, text, html }) => {
       text,
       html,
     };
-
+    console.time();
```
Comment (tdkent) [inline server/src/utils/emailHelper.js:102]: What do we need this for?

**Review by tdkent — APPROVED** [review]
Overall the search works well and good to merge as is. I have a couple of suggestions for how the UX might be improved:

- It's not really possible for the user to tell which columns can be filtered, or that all columns are being filtered by default. A label, placeholder, or tooltip explaining this might be worth adding.
- It might be helpful and clearer for the user to be able to select the column they want to filter by, instead of filtering all columns by default. For example I was trying to filter by name and it wasn't working like I expected because it was also filtering the email!
  - Side question: Would it helpful to force users to provide a first and last name in the request form, instead of using the `filterName` function?

File: server/src/utils/emailHelper.js (hunk 3 above)
Comment (kristi-h) [reply server/src/utils/emailHelper.js:102]: lol we don't.. I'll take it off.
I was testing something yesterday and that just snuck in with my other commits.

**Comment by kristi-h** [comment]
> * It's not really possible for the user to tell which columns can be filtered, or that all columns are being filtered by default. A label, placeholder, or tooltip explaining this might be worth adding.
I did have a placeholder that one point but that disappeared when I added the customized styling.
I like the idea of a tooltip! I'll add that.
> * It might be helpful and clearer for the user to be able to select the column they want to filter by, instead of filtering all columns by default. For example I was trying to filter by name and it wasn't working like I expected because it was also filtering the email!
Ah, well that's already available to do with the individual column filtering option. The searchbar filter is just the quick filter for all fields. But I get it's not clear for the user right now. Do you think the tool tip might help mitigate this confusion?
>   
>   * Side question: Would it helpful to force users to provide a first and last name in the request form, instead of using the `filterName` function? 
Sure~ I have no strong feelings about this.

**Pull request merged**